cni/network/plugin/main.go

```diff
@@ -259,10 +259,11 @@ func main() {
 	reflect.ValueOf(reportManager.Report).Elem().FieldByName("CniSucceeded").SetBool(true)
 	reflect.ValueOf(reportManager.Report).Elem().FieldByName("OperationDuration").SetInt(executionTimeMs)
 
-	if err = reportManager.SendReport(tb); err != nil {
-		log.Errorf("SendReport failed due to %v", err)
-	} else {
-		log.Printf("Sending report succeeded")
+	if cniReport.ErrorMessage != "" || cniReport.EventMessage != "" {
+		if err = reportManager.SendReport(tb); err != nil {
+			log.Errorf("SendReport failed due to %v", err)
+		} else {
+			log.Printf("Sending report succeeded")
```

Comment:
Shouldn't this be logged even if ErrorMessage/EventMessage is empty?

Reply:
When ErrorMessage and EventMessage are both empty, we don't want to send the report because it causes empty log statements in telemetry like [this](https://jarvis-west.dc.ad.msft.net/6753C8EF) (look at "Raw" column"

cniReport is a pointer to the report object inside of reportManager. reportManager.SendReport() is sends the cniReport object to telemetry. That's why were checking for cniReport.ErrorMessage and cniReport.EventMessage first before reportManager.SendReport().